Kentor.AuthServices/SAML2P/Saml2AuthenticationRequest.cs

```diff
@@ -84,6 +84,17 @@ private Saml2AuthenticationRequest(XmlDocument xml)
             {
                 AssertionConsumerServiceUrl = new Uri(AssertionConsumerServiceUriString);
             }
+
+            var nameIdPolicyElement = xml.DocumentElement["NameIDPolicy", Saml2Namespaces.Saml2PName];
+            if (nameIdPolicyElement == null) return;
+            
+            bool allowCreate;
+            bool.TryParse(nameIdPolicyElement.Attributes["AllowCreate"].GetValueIfNotNull(), out allowCreate);
```

Comment:
Check return value of TryParse. I want it to be strict - if the attribute is present it should be possible to parse. If not, throw an exception and reject the entire message.